ls1intum/artemis-android
# `Communication`: Add ability to pin messenges

### Description

This PR introduces pinning posts. A new post action has been created to allow users to pin standalone posts. Pinning posts is only available for channel creators and moderators, which is why the according flags are passed down to the post action. 

With the pinning feature added, it becomes quite clear that the post items in the chat need an overhaul to make the whole view coherent again. This will happen as a follow-up PR as described [here](https://github.com/ls1intum/artemis-android/issues/162).

### Changes

- The database had to be migrated as there was no column for the display priority in the Posts table.
- The PostItem was changed in order to give pinned posts a different look and show a pinned label on top of the post (see screenshot). 
- This PR includes multiple UI and E2E tests to ensure the correct behavior.
- MetisModificationService was changed to handle the modification request

### Steps for testing

1. Make sure you have moderation rights or create a chat before testing
2. Send a message and long click on it.
3. The `Pin Post` button should be visible.
4. Click the button and see how the message changes its appearance in the chat (A reload might be required because of ongoing issues with the websocket).
5. Long click again, click `Unpin Post` and verify that the message returns to its initial state
6. Verify that the Pin Post option is not available on answer posts in the thread view

### Screenshots
<img src="https://github.com/user-attachments/assets/2fb425ae-df36-4683-a97c-b70205fd7a15" width="500">

<img src="https://github.com/user-attachments/assets/0332b868-02e6-49fe-8dc1-fb49037315ae" width="350">

<img src="https://github.com/user-attachments/assets/600a9d81-414f-4d6a-bceb-696eb71480e9" width="350">
